### PR TITLE
Add a test for clickhouse as a logical replication consumer

### PR DESCRIPTION
**Pull request by a-masterov**
## Problem

We need to test logical replication with 3rd-party tools regularly. 

## Summary of changes

Added a test

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics? if so did you add the relevant metrics to the dashboard?
- [ ] If this PR requires public announcement, mark it with /release-notes label and add several sentences in this section.

## Checklist before merging

- [ ] Do not forget to reformat commit message to not include the above checklist

### CODE REVIEW

**Review by bayandin — COMMENTED** [review]
I think we need to create a separate test suite for external logical replication consumers, then we can put the test logic into the test itself instead of a bash file.

`test_pg_clients` can be used as an example (if needed).

**Review by bayandin — COMMENTED** [review]
Looks really nice!
Just one suggestion and a couple of comments 👍

File: poetry.lock
[the remark this answers is not in the document]
Comment (a-masterov) [reply poetry.lock:1]: Thank you,  done.

File: scripts/download_basebackup.py [hunk 1]
```diff
@@ -20,7 +20,7 @@ def main(args: argparse.Namespace):
     psconn: PgConnection = psycopg2.connect(pageserver_connstr)
     psconn.autocommit = True
 
-    output = open(output_path, "wb")
+    output = open(output_path, "w")
```
[the remark this answers is not in the document]
Comment (bayandin) [reply scripts/download_basebackup.py:23]: It doesn't fail on CI (for main and other PRs) and locally for me: ` ./scripts/pysync && poetry run mypy scripts/download_basebackup.py`. 
I suspect the failure was caused by unexpected changes in `poetry.lock`.

[the remark this answers is not in the document]
Comment (a-masterov) [reply scripts/download_basebackup.py]: the change rolled back

File: test_runner/logical_repl/test_log_repl.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+            client,
+            "select * from db1_postgres.table1 order by 1",
+            "9eba2daaf7e4d7d27ac849525f68b562ab53947d",
+        )
+        log.debug(res.result_rows)
+        log.debug("Sleeping before final checking if Neon is still alive")
+        time.sleep(3)
+        cur.execute("SELECT 1")
+        clickhouse_cleanup(test_output_dir)
+    except Exception as e:
```
[the remark this answers is not in the document]
Comment (a-masterov) [reply test_runner/logical_repl/test_log_repl.py]: done

File: test_runner/fixtures/neon_fixtures.py [hunk 3]
```diff
@@ -187,6 +187,23 @@ def versioned_pg_distrib_dir(pg_distrib_dir: Path, pg_version: PgVersion) -> Ite
     yield versioned_dir
 
 
+@pytest.fixture(scope="function")
+def clickhouse_instance(test_output_dir):
```
[the remark this answers is not in the document]
Comment (a-masterov) [reply test_runner/fixtures/neon_fixtures.py]: done

File: test_runner/fixtures/neon_fixtures.py [hunk 4]
```diff
@@ -187,6 +187,23 @@ def versioned_pg_distrib_dir(pg_distrib_dir: Path, pg_version: PgVersion) -> Ite
...
+    """
+    Startup and teardown a docker container with Clickhouse
+    """
+    log.info("Starting ClickHouse container")
+    cmd = ["docker", "run", "-d",  "-p", "9000:9000", "-p",
+           "8123:8123", "-h", "clickhouse", "--name", "clickhouse", "clickhouse/clickhouse-server"]
+    log.debug("start cmd: %s", ' '.join(cmd))
+    subprocess_capture(test_output_dir, cmd, check=True, capture_stdout=True)
```
[the remark this answers is not in the document]
Comment (a-masterov) [reply test_runner/fixtures/neon_fixtures.py]: Hmm, I cannot see it:
```
2024-07-24 17:37:36.269 INFO [test_log_repl.py:56] Starting ClickHouse container
2024-07-24 17:37:36.269 DEBUG [test_log_repl.py:71] start cmd: docker run -d -p 9000:9000 -p 8123:8123 -h clickhouse --name clickhouse clickhouse/clickhouse-server
2024-07-24 17:37:36.269 INFO [utils.py:124] Capturing stdout to "docker_1.stdout" and stderr to "docker_1.stderr"
```

[the remark this answers is not in the document]
Comment (bayandin) [reply test_runner/fixtures/neon_fixtures.py]: You're right, it prints the command to output files `docker_1.stdout` / `docker_1.stderr`

File: test_runner/logical_repl/test_log_repl.py [hunk 5]
```diff
@@ -3,18 +3,15 @@
...
 import clickhouse_connect
 import psycopg2
 import pytest
 from clickhouse_connect.driver.exceptions import DatabaseError
 from fixtures.log_helper import log
 from fixtures.neon_fixtures import RemotePostgres
-from fixtures.utils import subprocess_capture
 
 
 def timeout_query_clickhouse(
```
[the remark this answers is not in the document]
Comment (a-masterov) [reply test_runner/logical_repl/test_log_repl.py:18]:
```
    for i in range(number_of_iterations):
        try:
            res = func()
```
How can we pass arguments?

File: test_runner/fixtures/neon_fixtures.py [hunk 6]
```diff
@@ -187,6 +187,23 @@ def versioned_pg_distrib_dir(pg_distrib_dir: Path, pg_version: PgVersion) -> Ite
...
 
 
+@pytest.fixture(scope="function")
+def clickhouse_instance(test_output_dir):
+    """
+    Startup and teardown a docker container with Clickhouse
+    """
+    log.info("Starting ClickHouse container")
+    cmd = ["docker", "run", "-d",  "-p", "9000:9000", "-p",
+           "8123:8123", "-h", "clickhouse", "--name", "clickhouse", "clickhouse/clickhouse-server"]
```
[the remark this answers is not in the document]
Comment (a-masterov) [reply test_runner/fixtures/neon_fixtures.py]: done

File: test_runner/logical_repl/test_log_repl.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (bayandin) [reply test_runner/logical_repl/test_log_repl.py:18]: You can create a lambda function for it

[the remark this answers is not in the document]
Comment (a-masterov) [reply test_runner/logical_repl/test_log_repl.py]: done

File: .github/workflows/pg-clients.yml [hunk 7]
```diff
@@ -49,6 +49,72 @@ jobs:
...
+        env:
+          REGRESS_TEST_RESULT_CONNSTR_NEW: ${{ secrets.REGRESS_TEST_RESULT_CONNSTR_NEW }}
+
+      - name: Post to a Slack channel
+        if: github.event.schedule && failure()
+        uses: slackapi/slack-github-action@v1
+        with:
+          channel-id: "C06KHQVQ7U3" # on-call-qa-staging-stream
+          slack-message: |
+            Testing Postgres clients: <${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}|${{ job.status }}> (<${{ steps.create-allure-report.outputs.report-url }}|test report>)
```
Comment (bayandin) [inline .github/workflows/pg-clients.yml:114]: You need to update the message also

**Review by bayandin — COMMENTED** [review]
You need to update paths here as well:
```
  pull_request:
    paths:
      - '.github/workflows/pg-clients.yml'
      - 'test_runner/pg_clients/**'
      - 'poetry.lock'
```

File: .github/workflows/pg-clients.yml (hunk 7 above)
Comment (a-masterov) [reply .github/workflows/pg-clients.yml]: thank you, done

**Review by bayandin — APPROVED** [review]
Good work!

One more thing left to be done: https://github.com/neondatabase/neon/pull/8408#pullrequestreview-2199206023

**Pull request merged**